src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowTests.cs

```diff
@@ -4004,6 +4004,606 @@ public void M4(C? c)
+            ? x.ToString() // 1
+            : y.ToString(); // unreachable
+    }
+
+    void M2(C? c)
+    {
+        int x, y;
+        _ = c?.M0(x = y = 0) is C or null
+            ? x.ToString() // 2
+            : y.ToString(); // unreachable
+    }
+
+    void M3(C? c)
+    {
+        int x, y;
+        _ = c?.M0(x = y = 0) is not null and C
+            ? x.ToString()
+            : y.ToString(); // 3
+    }
+
+    void M4(C? c)
+    {
+        int x, y;
+        _ = c?.M0(x = y = 0) is null
+            ? x.ToString() // 4
+            : y.ToString();
+    }
+
+    void M5(C? c)
+    {
+        int x, y;
+        _ = c?.M0(x = y = 0) is not (C or { })
+            ? x.ToString() // 5
+            : y.ToString();
+    }
+
+    void M6(C? c)
+    {
+        int x, y;
+        _ = c?.M0(x = y = 0) is _ and C
+            ? x.ToString()
+            : y.ToString(); // 6
+    }
+
+    void M7(C? c)
+    {
+        int x, y;
+        _ = c?.M0(x = y = 0) is C and _
+            ? x.ToString()
+            : y.ToString(); // 7
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (13,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(13, 15),
+                // (21,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(21, 15),
+                // (30,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 3
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(30, 15),
+                // (37,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 4
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(37, 15),
+                // (45,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 5
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(45, 15),
+                // (54,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 6
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(54, 15),
+                // (62,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 7
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(62, 15)
+                );
+        }
+
+        [Fact]
+        public void IsCondAccess_05()
+        {
+            var source = @"
+#nullable enable
+
+class C
+{
+    int M0(object obj) => 1;
```

Comment:
>int [](start = 4, length = 3)

Consider a version where this returns `int?`.

Reply:
And specifically include a case where you test against `null`.

---
In reply to: [615140930](https://github.com/dotnet/roslyn/pull/52616#discussion_r615140930) [](ancestors = 615140930)

Reply:
Have expanded the test to cover these scenarios.